### PR TITLE
Client Runs: Runs Export

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Adding a new API endpoint to download multiple runs for a node. 

### :chains: Related Resources
#1764

### :athletic_shoe: How to Build and Test the Change
1. Add a button to the UI to download the file.
Add this code to the line below this link. 
https://github.com/chef/automate/blob/e1020ed008c7b94a42af91d8b0847cd6b58725ff/components/automate-ui/src/app/page-components/run-history/run-history.component.html#L23
```
      <chef-button class="dropdown-toggle" secondary (click)="onDownloadRunsReport()">
        <chef-icon>cloud_download</chef-icon>
      </chef-button>
```
1. build all the components and start Automate with `build components/config-mgmt-service && build components/automate-gateway build components/automate-ui && start_all_services`
1. Add some client run nodes with `chef_load_nodes 10`
1. Run the above line a few times to add multiple runs to a signal node. 
1. Open https://a2-dev.test/infrastructure/client-runs
1. Click on a node to open the node details page. 
1. Click the "Run history" button on the right of the page. Screen Shot 2019-10-14 at 3 50 28 PM
1. To download a CSV report, click the button that was created from the first step in the run history. 
Screen Shot 2019-10-14 at 3 55 07 PM

1. Find a node that has a successful and failure node and test filtering the runs and downloading a report. 

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: components/config-mgmt-service/integration_test/report_export_test.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+			runIsReturned:   false,
+		},
+		{
+			description:     "Run is returned when the run has a matching project with one of the allowed project with unassigned",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{authzConstants.UnassignedProjectID, "one"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when the run as a project and all projects are allowed",
```
Comment (msorens) [inline components/config-mgmt-service/integration_test/report_export_test.go]: Something missing--maybe "as" should be "has" ?

File: components/config-mgmt-service/integration_test/report_export_test.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when both of the two run projects match both of the allowed projects",
+			runsProjects:    []string{"one", "two"},
+			allowedProjects: []string{"one", "two"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is retuned when the run does not have an assigned project and the allowed project is unassigned",
```
Comment (msorens) [inline components/config-mgmt-service/integration_test/report_export_test.go]:
```suggestion
			description:     "Run is returned when the run does not have an assigned project and the allowed project is unassigned",
```

File: components/config-mgmt-service/integration_test/report_export_test.go (hunk 1 above)
Comment (lancewf) [reply components/config-mgmt-service/integration_test/report_export_test.go]: You are right, it was missing "has".

File: components/config-mgmt-service/integration_test/report_export_test.go (hunk 2 above)
Comment (lancewf) [reply components/config-mgmt-service/integration_test/report_export_test.go]: Thank you for taking a look. I updated the above problems.

File: .studio/automate-gateway [hunk 3]
```diff
@@ -228,17 +228,17 @@ function send_chef_run_failure_example() {
...
   local rfc_time
-  rfc_time=$(date +%FT%TZ -d "$((RANDOM % 144)) hour ago")
+  rfc_time=$(date +%FT%TZ -d "5 hour ago")
 
   local examples_path="/src/components/ingest-service/examples"
   [[ "$1" == "legacy" ]] && endpoint="/events/data-collector" || endpoint="/ingest/events/chef/run"
 
   # Update the IDs, names, and dates
-  tmp_ccr_json="$(jq --arg rfc_time "$rfc_time" --arg run_uuid "$run_uuid" --arg node_name "$node_name" --arg node_uuid "$node_uuid" '.end_time = $rfc_time | .start_time = $rfc_time | .entity_uuid = $node_uuid | .id = $run_uuid | .run_id = $run_uuid | .node_name = $node_name' <$examples_path/converge-failure-report.json)"
+  tmp_ccr_json="$(jq --arg rfc_time "$rfc_time" --arg run_uuid "$run_uuid" --arg node_name "$node_name" --arg node_uuid "$node_uuid" '.end_time = $rfc_time | .start_time = $rfc_time | .entity_uuid = $node_uuid | .id = $run_uuid | .run_id = $run_uuid | .node_name = $node_name' <$examples_path/converge-success-report.json)"
```
Comment (lancewf) [inline .studio/automate-gateway]: I am removing this.

File: components/automate-ui/src/app/page-components/run-history/run-history.component.ts [hunk 4]
```diff
@@ -106,6 +110,24 @@ export class RunHistoryComponent implements OnInit, OnDestroy {
     return moment.duration(moment(end_time).diff(moment(start_time))).humanize();
   }
 
+  onDownloadRunsReport() {
```
Comment (lancewf) [inline components/automate-ui/src/app/page-components/run-history/run-history.component.ts]: When the UI of the button is added this function can be used.

File: components/config-mgmt-service/grpcserver/report_export.go [hunk 5]
```diff
@@ -0,0 +1,280 @@
+package grpcserver
```
Comment (lancewf) [inline components/config-mgmt-service/grpcserver/report_export.go]: This file contains the main logic for this PR.

File: components/config-mgmt-service/backend/elastic/runs.go [hunk 6]
```diff
@@ -158,6 +158,56 @@ func (es Backend) GetRuns(nodeID string, page int, perPage int, filters map[stri
 	return runs, nil
 }
 
+func (es Backend) GetRunsPageByCurser(ctx context.Context, nodeID string, start time.Time,
```
Comment (lancewf) [inline components/config-mgmt-service/backend/elastic/runs.go]: This function allows walking through all the runs of a node with no 10,000 item limit.

**Comment by vjeffrey** [comment]
pulling this down to review now!

**Pull request merged**